tests/libcxx-backports/extents/CtorTestCombinations.h

```diff
@@ -45,15 +45,20 @@ constexpr void test_construction(AllExtents all_ext) {
 
   // test construction from just dynamic extents
   // create an array of just the extents corresponding to dynamic values
-  std::array<typename AllExtents::value_type, E::rank_dynamic()> dyn_ext{0};
-  size_t dynamic_idx = 0;
-  for (size_t r = 0; r < E::rank(); r++) {
-    if (E::static_extent(r) == std::dynamic_extent) {
-      dyn_ext[dynamic_idx] = all_ext[r];
-      dynamic_idx++;
+  if constexpr (E::rank_dynamic() > 0) {
```

Comment:
I don't understand.  What is going on here?

Reply:
It didn't like the init and assignment to the array in the case of length zero. There are likely other ways to rewrite this if so desired.